### PR TITLE
Create the 3.42 release post

### PR DESCRIPTION
**Pull request by iskyOS**
Creating the template blog post for the Sourcegraph 3.42 Release Post.

Sourcegraph 3.42 is planned to be released on July 22. 

EOD 7/18: Product adds highlights to the release
EOD 7/19: Release cut, changelog reviewed and added
7/20-21: Content + editorials review
7/22: Publish

### CODE REVIEW

**Review by ryphil — DISMISSED** [review]

File: content/blogposts/2022/release-3.42 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## New search suggestions
+We've moved to [CodeMirror](https://codemirror.net/) from Monaco for all our search query inputs. We're very excited about the performance and extensibility offered by CodeMirror and look forwaard to shipping all the improvements its flexibility enables. You can switch abck to using Monaco at any time by setting `experimentalFeatures.editor: "monaco"` in your settings menu.
+
+## Lucky search
+We have a new search type available experimentally, called "Lucky search". It does some really cool things to improve the quality of your search experience, but the most obvious one is that it gives smart suggestions for alternate queries, and runs them automatically to make sure they have results. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
```
Comment (Joelkw) [inline content/blogposts/2022/release-3.42:36]:
```suggestion
![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)

## Code Insights scoped to explicit repository lists load faster and include more historical datapoints

[Gif TODO – need to mock up old and new state next to each other]

Code insights running over an explicit list of repos now behave like code insights running over all repositories. 

They: 
- Backfill 12 historical datapoints (instead of 7)
- Now load 3-10x faster on future page visits thanks to persisting their datapoints
- Better handle large results sets and repositories
- Take daily snapshots of the current state and save it at your defined interval

## Code Insights better supports large monorepos 

Code insights now better supports running insights over large monorepos. Code insights now use our streaming endpoints to avoid being limited by global network timeouts, have improved retry behavior, and send insights running over explicitly-listed individual monorepos to our persistent backend (see the prior paragraph above). 
```

File: content/blogposts/2022/release-3.42 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## All-new Admin Analytics 
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders. 
+
+## Improved search-based navigation for Python non-locals
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:24]: @macraig What's the customer impact of this? Is it more accurate nav for certain use cases?

File: content/blogposts/2022/release-3.42 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+We previously announced we’d [improved our search-based support for Python locals in Sourcegraph 3.40(https://about.sourcegraph.com/blog/release/3.40) by using tree-sitter. We've now extended this improvement for jump-to-definition to more kinds of variables such as class fields.
+
+## Beta UI improvements
+We're fixing UX papercuts and cleaning up our UI. 3.42 will have an optional Beta UI, which can be enabled via a toggle in the avatar menu. It's disabled by default but we encourage you to give it a spin! We think you'll find it cleaner and easier to use. You can find the toggle just below the theme dropdown in the avatar menu.
+
+## Searching by repository descriptions
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## New search suggestions
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:34]: @benvenker What's the customer impact/benefit of this? Will it feel significantly different to the average user?

File: content/blogposts/2022/release-3.42 [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders. 
+
+## Improved search-based navigation for Python non-locals
+We previously announced we’d [improved our search-based support for Python locals in Sourcegraph 3.40(https://about.sourcegraph.com/blog/release/3.40) by using tree-sitter. We've now extended this improvement for jump-to-definition to more kinds of variables such as class fields.
+
+## Beta UI improvements
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:27]: @benvenker Is there any more narrative you can provide around why we decided to re-tune the UI, or the design principles that are being used to make the changes?

From an external perspective (just checking on it myself), it feels like the filters column just moved to the right, so I think it would help to provide more explanation to the user audience on what's happening there

File: content/blogposts/2022/release-3.42 (hunk 2 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42:24]: Since this is a shorter update, I wonder if this may fit better as a changelog item. We also have a lot of short updates in the highlights this month.

File: content/blogposts/2022/release-3.42 [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new Admin Analytics 
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders.
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:22]: @ryphil I'd suggest removing mention of license management for now and stick closer to what is part of this release — then can give the bigger picture in the full blog

File: content/blogposts/2022/release-3.42 [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders. 
+
+## Improved search-based navigation for Python non-locals
+We previously announced we’d [improved our search-based support for Python locals in Sourcegraph 3.40(https://about.sourcegraph.com/blog/release/3.40) by using tree-sitter. We've now extended this improvement for jump-to-definition to more kinds of variables such as class fields.
+
+## Beta UI improvements
+We're fixing UX papercuts and cleaning up our UI. 3.42 will have an optional Beta UI, which can be enabled via a toggle in the avatar menu. It's disabled by default but we encourage you to give it a spin! We think you'll find it cleaner and easier to use. You can find the toggle just below the theme dropdown in the avatar menu.
+
+## Searching by repository descriptions
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:30]: @benvenker, is this relevant to Self-hosted customers? This feels geared towards OSS search (on dotcom). Or is there a specific use case for self-hosted (mega enterprise with many repos maybe)?

File: content/blogposts/2022/release-3.42 (hunk 5 above)
Comment (ryphil) [reply content/blogposts/2022/release-3.42:22]: SGTM - Will remove.

File: content/blogposts/2022/release-3.42 (hunk 3 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:34]: @iskyOS it means users can search repositories by topic (as long as the repo description has topical keywords in it). It should feel significantly different to anyone who wants to find "types" of repos as opposed to a specific repo.

File: content/blogposts/2022/release-3.42 (hunk 4 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:27]: @iskyOS the major improvements so far are:
- filters moved to the right and are collapsible now
- removed excessive buttons in the header of the search results page
- search results are more compact and show fewer matches by default; they're also individually expandable with sticky headers and a sticky "show fewer" footer that is always visible, even if the results expand below the fold

This is part of a long term effort to clean up the core workflow (code search) and make it more focused and these changes are the first steps.

File: content/blogposts/2022/release-3.42 (hunk 6 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:30]: @iskyOS yeah if you only have a monorepo, less useful. If you're onboarding to a new team, company, or are already part of a large company with many repos, this means you don't have to know exactly what you're looking for, you can just describe the type of repo.

File: content/blogposts/2022/release-3.42 (hunk 2 above)
Comment (macraig) [reply content/blogposts/2022/release-3.42:24]: > Is it more accurate nav for certain use cases?
Yes, it's more accurate cross-file. It's already in the changelog so we can just remove it from this section if we already have a lot of updates.

File: content/blogposts/2022/release-3.42 (hunk 3 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42:34]: @benvenker, asking about CodeMirror implementation / new search suggestions

File: content/blogposts/2022/release-3.42 (hunk 2 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42:24]: OK — Let's have this one be a changelog item for this month.

File: content/blogposts/2022/release-3.42 [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:31]:
```suggestion
We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.

This is especially useful for engineers joining new teams or onboarding to a codebase at a new company, so you can quickly find repositories without knowing exactly what you're looking for.
```

Thoughts @benvenker ?

File: content/blogposts/2022/release-3.42 [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## New search suggestions
+We've moved to [CodeMirror](https://codemirror.net/) from Monaco for all our search query inputs. We're very excited about the performance and extensibility offered by CodeMirror and look forwaard to shipping all the improvements its flexibility enables. You can switch abck to using Monaco at any time by setting `experimentalFeatures.editor: "monaco"` in your settings menu.
+
+## Lucky search
+We have a new search type available experimentally, called "Lucky search". It does some really cool things to improve the quality of your search experience, but the most obvious one is that it gives smart suggestions for alternate queries, and runs them automatically to make sure they have results. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42:38]: @benvenker Are there other things this does we can mention, or point to in the docs?

If we say "it does some really cool things", I'd recommend giving customers the ability to see what those things are — otherwise we can drop that phrase and just talk about the smart suggestions

File: content/blogposts/2022/release-3.42 (hunk 7 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:30]: Great!

File: content/blogposts/2022/release-3.42 (hunk 3 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:33]: Whoops. For now, it means small improvements in speed and content in suggestions. Long term it will enable us to dramatically improve suggestion content.

File: content/blogposts/2022/release-3.42 (hunk 8 above)
Comment (benvenker) [reply content/blogposts/2022/release-3.42:37]: No docs yet. I'm fine with dropping the phrase.

File: content/blogposts/2022/release-3.42 [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+## New search suggestions
+We've moved to [CodeMirror](https://codemirror.net/) from Monaco for all our search query inputs. We're very excited about the performance and extensibility offered by CodeMirror and look forwaard to shipping all the improvements its flexibility enables. You can switch abck to using Monaco at any time by setting `experimentalFeatures.editor: "monaco"` in your settings menu.
+
+## Lucky search
+We have a new search type available experimentally, called "Lucky search". It improves the quality of your search experience by giving you smart suggestions for alternate queries, and it even runs them automatically to make sure they have results before recommending them. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
+![Search suggestion example](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
+
+## Code Insights scoped to explicit repository lists load faster and include more historical datapoints
+
+[Gif TODO – need to mock up old and new state next to each other]
```
Comment (Joelkw) [inline content/blogposts/2022/release-3.42:55]:
```suggestion
<p><video autoplay loop muted playsinline style="width:600px">
<source src="https://storage.googleapis.com/sourcegraph-assets/blog/3.42/3.42InsightsSpeedImprovements.mp4" type="video/mp4">
</video></p>


```

File: content/blogposts/2022/release-3.42.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  - description: 'Gitserver endpoint access logs can now be enabled for added visbility by adding "log": { "gitserver.accessLogs": true } to the config.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/
+    category: Admin
+---
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new admin analytics
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:32]:
```suggestion
We believe providing a world-class admin experience is critical for the success of Sourcegraph customers. This starts with enabling administrators to measure and track the value of Sourcegraph within their organization.
```

Changed admin-experience to admin experience. Also removed "around the world" - it felt repetitive with "world-class."

File: content/blogposts/2022/release-3.42.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization.
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:34]:
```suggestion
Historically, Sourcegraph administrators relied on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
```

File: content/blogposts/2022/release-3.42.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new admin analytics
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, identify power users, and convey value to their internal leaders.
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:36]: @iskyOS @ryphil - does "user analytics experience" suggest tracking devs? Should we just say "a redesigned analytics experience for administrators?"

Comment (abschu) [inline content/blogposts/2022/release-3.42.md:36]:
```suggestion
In 3.42, we are releasing a completely redesigned analytics experience for administrators. These visualizations will make it easier to understand developer engagement, identify power users, and measure the value of Sourcegraph. 
```

Comment (abschu) [reply content/blogposts/2022/release-3.42.md:36]: CC @Joelkw

File: content/blogposts/2022/release-3.42.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+## All-new admin analytics
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, identify power users, and convey value to their internal leaders. 
+
+![Admin analytics page](https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/admin_analytics.png)
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:38]: @iskyOS - a big portion of the image is the margins in the browser - not a blocker, but is it possible to show more of the UI and less of the margins?

File: content/blogposts/2022/release-3.42.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+## New search UI, now in Beta
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:40]: @benvenker @lcolston08  @iskyOS - this explains a lot about how to access the UI, but doesn't explain what is new about it. What specifically is new about it?

File: content/blogposts/2022/release-3.42.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+## Search by repository descriptions
+
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
+
+This is especially useful for engineers joining new teams or onboarding to a codebase at a new company, so you can quickly find repositories without knowing exactly what you're looking for.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## Smart search suggestions with lucky search
+
+We have a new search type available experimentally, called "Lucky search". It improves the quality of your search experience by giving you smart suggestions for alternate queries, and it even runs them automatically to make sure they have results before recommending them. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:54]:
```suggestion
We have a new search type available experimentally, called "Lucky search." It improves the quality of your search experience by giving you smart suggestions for alternate queries, and it even runs them automatically to make sure they have results before recommending them. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
```

File: content/blogposts/2022/release-3.42.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+![Search suggestion example](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
+
+## Code insights scoped to explicit repository lists load faster and include more historical datapoints
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:57]: Should Code Insights be capitalized?

Comment (abschu) [reply content/blogposts/2022/release-3.42.md:57]: @Joelkw @iskyOS - is this accurate:
Code Insights now load faster and include more historical data points

Is it necessary to say that it's scoped to explicit repository lists in the headline?

File: content/blogposts/2022/release-3.42.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+<video className="blog-image" title="Faster code insights" alt="An example of code insights loading faster" autoplay loop muted playsinline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/3.42/3.42InsightsSpeedImprovements.mp4" />
+</video>
+
+Code insights running over an explicit list of repos now behave like insights running over all repositories.
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:63]: @iskyOS - Did you decide to make all Code Insights references capitalized?

File: content/blogposts/2022/release-3.42.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+title: "Sourcegraph 3.42 release"
+publishDate: 2022-07-22T10:00-07:00
+description: "Sourcegraph 3.42 introduces new admin analytics, a new search UI in Beta, alternate search suggestions, and upgrades for code insights running on repo lists and monorepos."
+tags: [blog, release]
+slug: "release/3.42"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/sourcegraph-3-42.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/sourcegraph-3-42.png
+changelogItems:
+  - description: 'Further improved search-based code navigation for Python using tree-sitter. The upgrades to jump-to-definition have been extended to more kinds of variables such as class fields for more accurate cross-file navigation.'
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:11]: @iskyOS - should this be "We further improved...

File: content/blogposts/2022/release-3.42.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/38459
+    category: Search
+  - description: 'The homepage UI has been cleaned up, and panels such as recent searches are no longer visible by default. They can be re-enabled by setting "experimentalFeatures.showEnterpriseHomePanels" to true.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/37984
+    category: Search
+  - description: 'We have moved to CodeMirror from Monaco for all our search query inputs, resulting in faster and better quality search suggestions. You can switch back to Monaco by setting "experimentalFeatures.editor": "monaco" in the config.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/38584
+    category: Search
+  - description: 'Added experimental support for exporting traces to an OpenTelemetry collector by adding "observability.tracing": { "type": "opentelemetry" } to the config.'
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:20]: Will readers understand what this is/why it's valuable?

**Review by abschu — COMMENTED** [review]
@iskyOS - I reviewed and added comments.

File: content/blogposts/2022/release-3.42.md (hunk 19 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:20]: @abschu I believe so, yes. I can change the category to "Observability". Devs should understand tracing (and especially Eng Managers or DevEx folks) — it's a key part of service monitoring nowadays.

File: content/blogposts/2022/release-3.42.md (hunk 12 above)
Comment (ryphil) [reply content/blogposts/2022/release-3.42.md]: @abschu - Saw your suggestion below and accepted it. That was indeed confusing. We are releasing new analytics for *admins* to visualize usage of their instance.

**Review by ryphil — APPROVED** [review]

File: content/blogposts/2022/release-3.42.md (hunk 17 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:63]: @abschu, I was trying to do that, but I couldn't find a good way to rewrite this one to refer to the product rather than the charts. Do you think it should be caps even in the current context?

File: content/blogposts/2022/release-3.42.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+Historically, Sourcegraph administrators relied on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned analytics experience for administrators. These visualizations will make it easier to understand developer engagement, identify power users, and measure the value of Sourcegraph. 
+
+![Admin analytics page](https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/admin_analytics.png)
+
+## New search UI, now in Beta
+
+We're fixing UX papercuts and working to improve the core search experience. 3.42 includes a new, optional Beta UI, which can be enabled via a toggle in the avatar menu. It's disabled by default but we encourage you to give it a try. We think you'll find it cleaner and easier to use. You can find the toggle just below the theme dropdown in the avatar menu.
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42.md:42]:
```suggestion
We're fixing UX papercuts and working to improve the core search experience. 3.42 includes a new, optional Beta UI, which can be enabled via a toggle in the avatar menu. Filters have moved to the right and are now collapsible, search results are more compact and individually expandable, and buttons in the header have been combined to reduce clutter.

This UI is disabled by default but we encourage you to give it a try. We think you'll find it cleaner and easier to use. You can find the toggle below the theme dropdown in the avatar menu.
```

File: content/blogposts/2022/release-3.42.md (hunk 13 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:38]:
```suggestion
![Admin analytics page](https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/admin_analytics_v2.png)
```

File: content/blogposts/2022/release-3.42.md (hunk 16 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:57]: @abschu @Joelkw I think that change is fine, and the details can be in the body of the highlight.

File: content/blogposts/2022/release-3.42.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+Historically, Sourcegraph administrators relied on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned analytics experience for administrators. These visualizations will make it easier to understand developer engagement, identify power users, and measure the value of Sourcegraph. 
+
+![Admin analytics page](https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/admin_analytics_v2.png)
+
+## New search UI, now in Beta
+
+We're fixing UX papercuts and working to improve the core search experience. 3.42 includes a new, optional Beta UI, which can be enabled via a toggle in the avatar menu. It's disabled by default but we encourage you to give it a try. We think you'll find it cleaner and easier to use. You can find the toggle just below the theme dropdown in the avatar menu.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/beta-ui.gif)
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42.md:43]:
```suggestion

![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/beta-ui.gif)
```

File: content/blogposts/2022/release-3.42.md [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+
+## Search by repository descriptions
+
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
+
+This is especially useful for engineers joining new teams or onboarding to a codebase at a new company, so you can quickly find repositories without knowing exactly what you're looking for.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42.md:50]:
```suggestion

![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
```

File: content/blogposts/2022/release-3.42.md [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+
+## Smart search suggestions with lucky search
+
+We have a new search type available experimentally, called "Lucky search." It improves the quality of your search experience by giving you smart suggestions for alternate queries, and it even runs them automatically to make sure they have results before recommending them. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
+![Search suggestion example](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42.md:55]:
```suggestion

![Search suggestion example](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
```

File: content/blogposts/2022/release-3.42.md (hunk 22 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:50]:
```suggestion

![Search for repository by description](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
```

File: content/blogposts/2022/release-3.42.md (hunk 16 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:60]:
```suggestion
## Insights load faster and include more historical datapoints
```

File: content/blogposts/2022/release-3.42.md (hunk 17 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:66]:
```suggestion
Insights running over an explicit list of repos now behave like insights running over all repositories. 
```

File: content/blogposts/2022/release-3.42.md [hunk 24]
```diff
@@ -0,0 +1,78 @@
+---
+title: "Sourcegraph 3.42 release"
+publishDate: 2022-07-22T10:00-07:00
+description: "Sourcegraph 3.42 introduces new admin analytics, a new search UI in Beta, alternate search suggestions, and upgrades for code insights running on repo lists and monorepos."
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:4]: @iskyOS - Should Code Insights be capitalized here?

File: content/blogposts/2022/release-3.42.md [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+tags: [blog, release]
+slug: "release/3.42"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/sourcegraph-3-42.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/release-post/3.42/sourcegraph-3-42.png
+changelogItems:
+  - description: 'We further improved search-based code navigation for Python using tree-sitter. The upgrades to jump-to-definition have been extended to more kinds of variables such as class fields for more accurate cross-file navigation.'
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:11]:
```suggestion
  - description: 'We further improved search-based code navigation for Python using tree-sitter. The upgrades to jump-to-definition have been extended to more kinds of variables, such as class fields, for more accurate cross-file navigation.'
```

File: content/blogposts/2022/release-3.42.md [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+    url: https://github.com/sourcegraph/sourcegraph/pull/37984
+    category: Admin
+  - description: 'Gitserver endpoint access logs can now be enabled for added visbility by adding "log": { "gitserver.accessLogs": true } to the config.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/
+    category: Admin
+---
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new admin analytics
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:30]: Measure the value of Sourcegraph with admin analytics

File: content/blogposts/2022/release-3.42.md [hunk 27]
```diff
@@ -0,0 +1,78 @@
...
+
+![Search for repository by description](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## Smart search suggestions with lucky search
+
+We have a new search type available experimentally, called "Lucky search." It improves the quality of your search experience by giving you smart suggestions for alternate queries, and it even runs them automatically to make sure they have results before recommending them. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
+
+![Search suggestion example](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
+
+## Insights load faster and include more historical datapoints
```
Comment (abschu) [inline content/blogposts/2022/release-3.42.md:62]: For the headline, this should be "Code Insights" - sorry I misunderstood this on Slack.

**Review by abschu — COMMENTED** [review]
Looks good! Just a few minior comments.

File: content/blogposts/2022/release-3.42.md (hunk 27 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:62]:
```suggestion
## Code Insights load faster and include more historical datapoints
```

File: content/blogposts/2022/release-3.42.md (hunk 26 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:30]:
```suggestion
## Measure the value of Sourcegraph with admin analytics
```

File: content/blogposts/2022/release-3.42.md (hunk 24 above)
Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:4]: Yes! Good catch — saw this earlier but forgot to change.

Comment (iskyOS) [reply content/blogposts/2022/release-3.42.md:4]:
```suggestion
description: "Sourcegraph 3.42 introduces new admin analytics, a new search UI in Beta, alternate search suggestions, and upgrades for Code Insights running on repo lists and monorepos."
```

File: content/blogposts/2022/release-3.42.md [hunk 28]
```diff
@@ -0,0 +1,78 @@
+---
+title: "Sourcegraph 3.42 release"
+publishDate: 2022-07-22T10:00-07:00
```
Comment (iskyOS) [inline content/blogposts/2022/release-3.42.md:3]:
```suggestion
publishDate: 2022-07-25T10:00-07:00
```

**Review by ryphil — APPROVED** [review]
LGTM
